pkg/spanconfig/spanconfigsqltranslator/protectedts_state_reader.go

```diff
@@ -70,22 +75,32 @@ func (p *protectedTimestampStateReader) getProtectionPoliciesForSchemaObject(
 	return p.schemaObjectProtections[descID]
 }
 
-func (p *protectedTimestampStateReader) loadProtectedTimestampRecords(ptsState ptpb.State) {
+func (p *protectedTimestampStateReader) loadProtectedTimestampRecords(
+	ctx context.Context, jr *jobs.Registry, txn *kv.Txn, ptsState ptpb.State,
+) error {
 	tenantProtections := make(map[roachpb.TenantID][]roachpb.ProtectionPolicy)
 	for _, record := range ptsState.Records {
+		// TODO(adityamaru): When schedules start writing protected timestamp
+		// records extend this logic to also include those records.
+		isBackupProtectedTimestampRecord, err := jobsprotectedts.IsRecordWrittenByJobType(ctx, jr, txn,
```

Comment:
Huh, so recently I haven't followed everything here all that closely, so maybe this is a silly question, but it seems weird to me to go by who created the pts record rather than just have that creator set a bit in the record it creates and then look at that bit?

Reply:
I don't really like looking up the job here much. Is there not a proto in that table we can just add a field to vs a whole new column?

Reply:
I could stick it in https://github.com/cockroachdb/cockroach/blob/5a18c3ab8f4d1d096bc3a2d98bbe712d547c9cd1/pkg/kv/kvserver/protectedts/ptpb/protectedts.proto#L140. I guess it sorta fits that its information needed when evaluating the pts record and what it targets.

Reply:
We don't write the Record as a proto unfortunately, we pick out its fields.

```
   id        UUID NOT NULL,
   ts        DECIMAL NOT NULL,
   meta_type STRING NOT NULL,
   meta      BYTES,
   num_spans INT8 NOT NULL, -- num spans is important to know how to decode spans
   spans     BYTES NOT NULL,
   verified  BOOL NOT NULL DEFAULT (false),
   target    BYTES,         -- target is an encoded protobuf that specifies what the pts record will protect
```

Reply:
I think I like the idea of sticking it into `ptpb.Target`. Will rework this PR to do that instead.